tests/serverpod_test_server/test_integration/authentication/auth_config_changes_test.dart

```diff
@@ -0,0 +1,74 @@
+import 'package:serverpod_auth_server/module.dart';
+import 'package:serverpod_test_server/test_util/test_serverpod.dart';
+import 'package:test/test.dart';
+
+void main() async {
+  var userName = 'test';
+  var email = 'test@serverpod.dev';
+  var password = 'password';
+  var session = await IntegrationTestServer().session();
+
+  var codeLengthsToTest = [-16, -8, -4, -1, 0, 1, 4, 16];
+
+  test(
+      'Not setting validationCodeLength will generate a code with default length 8',
+      () async {
+    String? generatedValidationCode;
+    AuthConfig.set(
+      AuthConfig(
+        sendValidationEmail: (session, email, validationCode) async {
+          generatedValidationCode = validationCode;
+          return true;
+        },
+      ),
+    );
+
+    var createAccountRequest =
+        await Emails.createAccountRequest(session, userName, email, password);
+
+    expect(
+      createAccountRequest,
+      isTrue,
+      reason: 'Generated validation code is 8 characters long, which is valid',
+    );
+
+    assert(generatedValidationCode?.length == 8);
+  });
+
+  for (var length in codeLengthsToTest) {
+    final willPass = length >= 1;
+    test(
```

Comment:
Suggestion: Please break these out into individual explicit tests.

In tests it is fine to have code duplication, it is all about making sure the developer reading the test understand what is being tested and why.

I also think that it should be enough with three tests, one using a negative value, one using zero and one using a positive value :)